visualization/app/codeCharta/ui/codeMap/threeViewer/threeSceneService.ts

```diff
@@ -76,47 +82,42 @@ export class ThreeSceneService implements CodeMapPreRenderServiceSubscriber {
 		this.getMapMesh().highlightBuilding(this.highlighted, this.selected, state, this.constantHighlight)
 	}
 
-	private updateCorrectMaterial(materials: Material[], nodes: CodeMapBuilding[], select: boolean) {
-		// TODO This loops introduce performance issues
-		//  refactor it please
-		return
-		const allNodes = this.mapMesh.getNodes()
-
-		for (const node of allNodes) {
-			for (const material of materials) {
-				for (const hoveredNode of nodes) {
-					if (hoveredNode !== null) {
-						if (select) {
-							if (material.userData.id === hoveredNode.node.id) {
-								material["color"].setHex(ColorConverter.convertHexToNumber(hoveredNode.color))
-							}
-						} else if (material.userData.id === hoveredNode.node.id) {
-							material["color"].setHex(ColorConverter.convertHexToNumber("#FFFFFF"))
-						} else if (material.userData.id === node.id) {
-							material["color"].setHex(ColorConverter.convertHexToNumber("#7A7777"))
-						}
-					}
-				}
+	private selectMaterial(materials: Material[]) {
+		for (const material of materials) {
+			if (material.userData.id === this.selected.node.id) {
+				material["color"].setHex(this.numberOrangeColor)
 			}
 		}
```

Comment:
```suggestion
		const selectedMaterial = materials.find(({ userData }) => userData.id === this.selected.node.id)
		selectedMaterial?.["color"].setHex(this.numberOrangeColor)
```

Reply:
Could you explain the suggestion please, I don't think I understand your proposal.

Reply:
AFAIK only a single id is selected. As such, we only have to find that specific material. Using `.find()` is a convenient way to do that as it's similar to adding a `break` at the end of the if statement.
The second line executes the `.setHex()` function for the selected material in case it's not null or undefined.

It's therefore shorter and faster code :-)

Reply:
And thank you for the detailed explanation